ProjectEureka/project-eureka-flutter
# PE-85  | Connect More Details to Chat

What I Did:

- Connected The More details page to the chat system
- When clicking the more details page button on a question, you will be taken to the more details page. If you are not the user who answered the question, you should see an option in which you can answer the button with chat or no chat. Upon clicking chat, you will be brought to the chat screen between you and the user of the question. 
![Screen Shot 2021-04-20 at 11 40 39 AM](https://user-images.githubusercontent.com/48227740/115447333-3d13af80-a1cd-11eb-85a5-47c086ba651d.png)
![Screen Shot 2021-04-20 at 11 40 55 AM](https://user-images.githubusercontent.com/48227740/115447359-46048100-a1cd-11eb-8fe1-1a3d7948b0e3.png)
![Screen Shot 2021-04-20 at 11 41 44 AM](https://user-images.githubusercontent.com/48227740/115447461-63d1e600-a1cd-11eb-9cb5-350ab6bc5faa.png)
- If you are the user who asked the question, you will see the option of archiving the question. 
![Screen Shot 2021-04-20 at 11 43 40 AM](https://user-images.githubusercontent.com/48227740/115447702-a7c4eb00-a1cd-11eb-9fd1-53019bbbaaa4.png)

How To Test:

- Git Fetch, then Git Pull. Checkout into my branch PE-85 and start your emulator. 
- Upon logging in, you should see a question labeled "How to test Chat System". Find this question and click more details
![Screen Shot 2021-04-20 at 11 45 27 AM](https://user-images.githubusercontent.com/48227740/115447932-e8246900-a1cd-11eb-9697-c0980718f843.png)

- Start a chat via the answer button and send a message. 
- To confirm that this is working properly, let me know via discord so I can confirm that the message has been sent to the user. 
- Navigate to the Homepage and click the chat icon on the upper right hand side. You should see the conversation with the question title "How to test Chat System. Upon clicking on this, you will see the chat history.
- If all of this can be done, then the chat system is successful. 
![Screen Shot 2021-04-20 at 11 48 17 AM](https://user-images.githubusercontent.com/48227740/115448225-4cdfc380-a1ce-11eb-87fb-f5e1b003de5c.png)
![Screen Shot 2021-04-20 at 11 48 30 AM](https://user-images.githubusercontent.com/48227740/115448249-549f6800-a1ce-11eb-881d-d1e079a93ab8.png)
